CMakeLists.txt

```diff
@@ -31,8 +42,39 @@ file(GLOB_RECURSE HIPIFY_HEADERS src/*.h)
 add_llvm_executable(hipify-clang ${HIPIFY_SOURCES} ${HIPIFY_HEADERS})
 target_link_directories(hipify-clang PRIVATE ${LLVM_LIBRARY_DIRS})
 
-set(CMAKE_CXX_COMPILER ${LLVM_TOOLS_BINARY_DIR}/clang++)
-set(CMAKE_C_COMPILER ${LLVM_TOOLS_BINARY_DIR}/clang)
+set(HIPIFY_INSTALL_ROOT_DIR ${CMAKE_INSTALL_PREFIX})
+set(HIPIFY_INSTALL_BIN_DIR ${HIPIFY_INSTALL_ROOT_DIR/bin})
```

Comment:
I still do not understand why the existing path is changed and/or refactored. I believe it should be left untouched. Then, we can refactor it, if needed.

Reply:
That is not true. I am doing monolithic build, so that works fine.

Reply:
I have dropped all of the cleanups and refactoring and left the bare minimum to make it "includable" in the HIP SDK. Feel free to do those cleanups and refactoring later.

I had to change the **install** command for hipify-perl:

`DIRECTORY ${CMAKE_SOURCE_DIR}/bin -> DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bin`

it works "as is" if you build hipify standalone, but it is incorrect to use CMAKE_SOURCE_DIR if you build it as part of other project, because CMAKE_SOURCE_DIR may point to other project, however CMAKE_CURRENT_SOURCE_DIR will point to the current source directory...